### ISSUE TITLE
Thread Sanitizer: Swift access race in Segment.CountBasedFlushPolicy.count.modify : Swift.Int at 0x11a75f4c0

### ISSUE THREAD

**Issue by dmcgloin** [segmentio/analytics-swift#227]
**Describe the bug**

When running our app with the Xcode Thread Sanitizer enabled, we sometimes see this runtime warning: `Swift access race in Segment.CountBasedFlushPolicy.count.modify : Swift.Int at 0x11a75f4c0`

It's flagged within the `CountBasedFlushPolicy` class at this line of code:

```
   public func updateState(event: RawEvent) {
        count += 1      <-----
    }
```

**To Reproduce**

1. Edit the current main app target (iOS) scheme
2. For the Run (Debug) section, select Diagnostics, enable "Thread Sanitizer" check box
3. Run app

**Expected behavior**
No runtime warnings

**Platform (please complete the following information):**
 - Library Version in use: 1.4.5 (But I don't think this issue is new to this version)
 - Platform being tested: iOS 16.4 on simulator
 - Integrations in use: None except custom proxy

**Comment by dmcgloin** [comment]
FWIW, also just saw a new Thread Sanitizer warning with same release:

`Swift access race in Segment.StartupQueue.running.getter : Swift.Bool at 0x11db41080`

...at this code:

```
    public func execute<T: RawEvent>(event: T?) -> T? {
        if running == false, let e = event  {      <-----
            // timeline hasn't started, so queue it up.
```

**Comment by dannys42** [comment]
Also seen in `reset()`:

```
    public func reset() {
        count = 0
    }
```

**Comment by bsneed** [comment]
Thanks guys!  Just got back into the office today and will jump on this!  <highfive>

**Comment by bsneed** [comment]
> FWIW, also just saw a new Thread Sanitizer warning with same release:
> 
> `Swift access race in Segment.StartupQueue.running.getter : Swift.Bool at 0x11db41080`
> 
> ...at this code:
> 
> ```
>     public func execute<T: RawEvent>(event: T?) -> T? {
>         if running == false, let e = event  {      <-----
>             // timeline hasn't started, so queue it up.
> ```

Hmm, this one is interesting, as `running` is being hit via an accessor that uses nslock to acquire the value.  I wonder if it's a false alarm.

**Comment by bsneed** [comment]
Fixed!  Thanks for this everyone, I feel like it was a pretty significant improvement overall having looked at it with a TSan lens.

Another bug to fix, and then this will get released.

**Comment by bsneed** [comment]
Fixed in 1.4.6
